### PR TITLE
docs: add README for `upgrade-testing.yml` workflow

### LINKED ISSUE #770: Add README for `upgrade-testing.yml` workflow
### Problem

There are no examples of how devs should be able to use this workflow and what use cases could be covered.

### Solution

- rename `upgrate-testing.yml` to `migration-testing.yml` and follow up names (in terms of the comments in the main branch [here](https://github.com/hashgraph/hedera-smart-contracts/pull/757))
- add detailed README

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
**Description**:

There are no examples of how devs should be able to use this workflow and what use cases could be covered.

**Solution**:

- rename `upgrate-testing.yml` to `migration-testing.yml` and follow up names (in terms of the comments in the main branch [here](https://github.com/hashgraph/hedera-smart-contracts/pull/757))
- add detailed README

**Related issue(s)**:

Fixes #770 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by JeffreyDallas — APPROVED** [review]

File: .github/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Hedera Smart Contracts Workflows
+
+A detailed overview of how you can use hedera-smart-contract's github workflows for testing purposes. Besides examples, there will be explanations of what is happening under the hood.
+
+## Actions
+
+### Migration Testing
+
+When there is a new version released, no matter if it is network-node, mirror-node, or relay, we want to make sure the entire ecosystem is working as expected and there are not any critical regressions introduced. Another concern with switching the network node's versions is "the migration" that is running under the hood. It occurs every time a node has been started and detects if there is data for migration. The migration testing workflow offers this type of testing where we can choose versions for pre/post image tags.
```
Comment (quiet-node) [inline .github/README.md:9]: I feel like we can drop the `Another concern with switching...` to a new paragraph for easier reading.

File: .github/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- **initialRelayTag** - Specify the initial Relay image tag
+- **targetNetworkNodeTag** - Specify target Network Node image tag
+- **targetMirrorNodeTag** - Specify the target Mirror Node image tag
+- **targetRelayTag** - Specify the target Relay image tag
+- **preMigrationTestTags** - Specify the pre-migration test tags. Default: @pre-migration. It could be every tag we want (e.g. **@OZERC20**)
+- **postMigrationTestTags** - Specify the post-migration test tags. Default: @post-migration. It could be every tag we want (e.g. **@OZERC20**)
+
+Examples:
+- if we want to test possible state corruption between mono and mod versions, we could use a setup like this:
+    - initialNetworkNodeTag: 0.48.1
```
Comment (quiet-node) [inline .github/README.md:23]: Maybe add a comment-type explaination saying this is the last mono image tag, or something like that
``` suggestion
    - initialNetworkNodeTag: 0.48.1 # last tag for mono module
```

File: .github/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - initialMirrorNodeTag: 0.103.0
+    - initialRelayTag: 0.47.0
+    - targetNetworkNodeTag: 0.49.7
```
Comment (quiet-node) [inline .github/README.md:26]: Maybe add a comment-type explaination saying this is the first stable tag for modular module, or something like that
```suggestion
    - targetNetworkNodeTag: 0.49.7 # first stable tag for modular module
```

**Review by quiet-node — COMMENTED** [review]
Nice work! Leaving some comments.

**Review by quiet-node — DISMISSED** [review]
LGTM

**Review by isavov — APPROVED** [review]

**Pull request merged**